compose/service.py

```diff
@@ -402,11 +402,15 @@ def config_hash(self):
         return json_hash(self.config_dict())
 
     def config_dict(self):
-        return {
+        dict = {
             'options': self.options,
-            'image_id': self.image()['Id'],
         }
 
+        if self.image() is not None:
+            dict['image_id'] = self.image()['Id']
```

Comment:
How about:

```python

return {
    ...
    'image_id': (self.image() or {}).get('Id')
}
```

That way the API call is only performed once, and I think it would be the same behaviour